web/src/client/network.js

```diff
@@ -212,7 +212,8 @@ class NetworkManagerAdapter {
     // FIXME: find a better way to add gateway only if there are addresses. If not, a DBusError will
     // be raises "gateway cannot be set if there are no addresses configured".
     if (newSettings.ipv4["address-data"].v.length !== 0) {
-      newSettings.ipv4.gateway = cockpit.variant("s", connection.ipv4.gateway);
+      const gateway = newSettings.ipv4.method === "manual" ? newSettings.ipv4.gateway : "";
```

Comment:
I guess it is not needed here

Reply:
Why? Could you please tell a bit more?

Reply:
Sorry, my fault, I was thinking it was in other part of the code looks good ;)

Reply:
Even with DHCP we could set the gateway and also maintain the additional addresses, for example if you do not get a DHCP lease, you will use the gateway defined... so maybe we should just allow to set the field unconditionally as we do with additional IP addresses.

Reply:
Therefore I would do `newSettings.ipv4.gateway = cockpit.variant("s", newSettings.ipv4.gateway)`  unconditionally.

Reply:
Therefore I would not touch the gateway here at all, and just use what is in newSettings.ipv4